superset/connectors/druid/models.py

```diff
@@ -361,21 +361,18 @@ def get_metrics(self):
                 )
         return metrics
 
-    def generate_metrics(self):
-        """Generate metrics based on the column metadata"""
-        metrics = self.get_metrics()
-        dbmetrics = (
-            db.session.query(DruidMetric)
```

Comment:
Running a loop to issue a query for each metric/column means that many queries have to be made, as opposed to just one to grab them all at once. If you've got tons of metrics and such, this adds up and can be reasonably slow

Reply:
@Mogball that was one of my concerns and I agree with your comment. I've refactored the logic to do one query per `DruidColumn`.